CONTRIBUTING.md

```diff
@@ -1,294 +1,327 @@
+mkdir -p registry/[your-username]/.images
+```
 
-> [!WARNING]
-> These instructions cannot be followed just yet; the script referenced will be made available shortly. Contributors looking to add modules early will need to create all directories manually.
+#### Add Your Avatar
 
-Once Bun (and possibly Go) have been installed, clone the Coder Registry repository. From there, you can run this script to make it easier to start contributing a new module or template:
+Every namespace must have an avatar. Add your avatar image:
 
-```shell
-./new.sh USER_NAMESPACE/NAME_OF_NEW_MODULE
-```
+1. Add a square image (recommended: 400x400px minimum) to `registry/[your-username]/.images/`
+2. Name it exactly `avatar.png` or `avatar.svg`
+3. Supported formats: `.png` or `.svg` only
 
-You can also create a module file manually by creating the necessary files and directories.
+#### Create Your Namespace README
 
-### The composition of a Coder module
+Create `registry/[your-username]/README.md`:
 
-Each Coder Module must contain the following files:
+```markdown
+---
+display_name: "Your Name"
+bio: "Brief description of who you are and what you do"
+avatar_url: "./.images/avatar.png"
+github: "your-username"
+linkedin: "https://www.linkedin.com/in/your-username" # Optional
+website: "https://yourwebsite.com" # Optional
+support_email: "you@example.com" # Optional
+status: "community"
+---
 
-- A `main.tf` file that defines the main Terraform-based functionality
-- A `main.test.ts` file that is used to validate that the module works as expected
-- A `README.md` file containing required information (listed below)
+# Your Name
 
-You are free to include any additional files in the module, as needed by the module. For example, the [Windows RDP module](https://github.com/coder/registry/tree/main/registry/coder/modules/windows-rdp) contains additional files for injecting specific functionality into a Coder Workspace.
+Brief description of who you are and what you do.
+```
 
-> [!NOTE]
-> Some legacy modules do not have test files defined just yet. This will be addressed soon.
+> **Note**: The `avatar_url` must point to `./.images/avatar.png` or `./.images/avatar.svg`.
 
-### The `main.tf` file
+### 2. Generate Module Files
 
-This file defines all core Terraform functionality, to be mixed into your Coder workspaces. More information about [Coder's use of Terraform can be found here](https://coder.com/docs/admin/templates/extending-templates/modules), and [general information about the Terraform language can be found in the official documentation](https://developer.hashicorp.com/terraform/docs).
+```bash
+./scripts/new_module.sh [your-username]/[module-name]
+cd registry/[your-username]/modules/[module-name]
+```
 
-### The structure of a module README
+This script generates:
 
-Validation criteria for module README files is listed below.
+- `main.tf` - Terraform configuration template
+- `README.md` - Documentation template with frontmatter
```

Comment:
This should also generate a `run.sh` file, which the module author can delete if not required.